src/west/app/main.py

```diff
@@ -90,9 +91,11 @@ def run(self, argv):
 
         # Read the configuration files. We need this to get
         # manifest.path to parse the manifest, etc.
-        #
-        # TODO: re-work to avoid global state (#149).
-        config.read_config(topdir=self.topdir)
+        self.config = west.configuration.Configuration(topdir=self.topdir)
+
+        # Also set up the global configuration object to match, for
```

Comment:
```suggestion
        # Also set up the deprecated global configuration object to match, for
```
Is it actually deprecated? Not sure yet.